one_fm/api/tasks.py

```diff
@@ -121,17 +130,23 @@ def notify(recipients,log_type):
 	checkout_subject = _("Final Reminder: Please checkout in the next five minutes.")
 	checkout_message = _("""<a class="btn btn-danger" href="/desk#face-recognition">Check Out</a>""")
 	Notification_title = "Final Reminder"
-	user_id = []
-	employee_id = []
+	Notification_body = "Please checkin in the next five minutes."
+	
+	#eg: recipient: ['s.shaikh@armor-services.com', 'HR-EMP-00001', 'Develope-Head Office-Morning-1']
 	for recipient in recipients:
-		user_id.append(recipient[0])
-		employee_id.append(recipient[1])
-	if log_type=="IN":
-		send_notification(checkin_subject, checkin_message, user_id)
-		push_notification(employee_id, Notification_title, "Please checkin in the next five minutes.")
-	if log_type=="OUT":
-		send_notification(checkout_subject, checkout_message, user_id)
-		push_notification(employee_id, Notification_title, "Please checkout in the next five minutes.")
+		user_id = recipient[0]
+		employee_id = recipient[1]
+		print(recipient)
+		if log_type=="IN":
+			send_notification(checkin_subject, checkin_message, user_id)
+			#arrive late button is true only if the employee is assigned with shift to head office.
+			if recipient[2] == "Head Office-Head Office-Morning|08:00:0-17:00:0|9 hours":

```

Comment:
Avoid hard coding here. Try using a different approach to set the default shift for head office and fetch that shift from where you set it.
Changing the name of the shift in future will cause problems.

Reply:
Used User Role instead.